### PR TITLE
Extract a Listenable base class from Animation and ChangeNotifier

### PR DESCRIPTION
**Pull request by abarth**
Having this base class lets classes like CustomPainter and DataTableSource be
more agnostic as to what's generating the repaints.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

**Review by dragostis — APPROVED** [review]
LGTM.

File: packages/flutter/lib/src/material/time_picker.dart [hunk 1]
```diff
@@ -426,6 +426,12 @@ class _DialState extends State<_Dial> {
       _animateTo(_getThetaForTime(config.selectedTime));
   }
 
+  @override
+  void dispose() {
+    _thetaController.dispose();
```
Comment (Hixie) [inline packages/flutter/lib/src/material/time_picker.dart]: !

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/material/time_picker.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/material/time_picker.dart]: Yeah, just happened to notice that one.

**Pull request merged**